File: backend/src/main/java/com/vueadmin/entity/system/Organization.java
package com.vueadmin.entity.system;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.Comment;

import java.time.LocalDateTime;

/**
 * 组织机构实体
 */
@Data
@Entity
@Table(name = "sys_organization")
@Comment("组织机构表")
public class Organization {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Comment("主键ID")
    private Long id;

    @Column(name = "org_code", nullable = false, unique = true, length = 100)
    @Comment("组织编码")
    private String orgCode;

    @Column(name = "org_name", nullable = false, length = 200)
    @Comment("组织名称")
    private String orgName;

    @Column(name = "org_type", length = 50)
    @Comment("组织类型: company/department/group/position")
    private String orgType;

    @Column(name = "parent_id")
    @Comment("父级ID")
    private Long parentId;

    @Column(name = "level")
    @Comment("层级")
    private Integer level;

    @Column(name = "path", length = 500)
    @Comment("路径")
    private String path;

    @Column(name = "manager", length = 50)
    @Comment("负责人")
    private String manager;

    @Column(name = "phone", length = 20)
    @Comment("电话")
    private String phone;

    @Column(name = "email", length = 100)
    @Comment("邮箱")
    private String email;

    @Column(name = "sort")
    @Comment("排序")
    private Integer sort;

    @Column(name = "status", length = 20)
    @Comment("状态: active/inactive")
    private String status;

    @Column(name = "created_by", length = 50)
    @Comment("创建人")
    private String createdBy;

    @Column(name = "created_at")
    @Comment("创建时间")
    private LocalDateTime createdAt;

    @Column(name = "updated_by", length = 50)
    @Comment("更新人")
    private String updatedBy;

    @Column(name = "updated_at")
    @Comment("更新时间")
    private LocalDateTime updatedAt;

    @Column(name = "description", columnDefinition = "TEXT")
    @Comment("描述")
    private String description;

    @PrePersist
    public void prePersist() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
        if (updatedAt == null) {
            updatedAt = LocalDateTime.now();
        }
        if (status == null) {
            status = "active";
        }
        if (sort == null) {
            sort = 0;
        }
        if (level == null) {
            level = 1;
        }
    }

    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/api/system/organization.ts
import api from '../index'

/**
 * 组织机构接口
 */

// 组织机构类型
export interface Organization {
  id: number
  orgCode: string
  orgName: string
  orgType: string
  parentId: number | null
  parentName?: string
  level: number
  path: string
  manager: string
  phone: string
  email: string
  sort: number
  status: string
  createdBy: string
  createdAt: string
  updatedBy: string
  updatedAt: string
  description: string
  children?: Organization[]
  hasChildren?: boolean
  label?: string
  value?: number
}

// 组织表单类型
export interface OrganizationForm {
  id?: number
  orgCode: string
  orgName: string
  orgType: string
  parentId: number | null
  manager: string
  phone: string
  email: string
  sort: number
  status: string
  description: string
  createdBy?: string
  updatedBy?: string
}

// 组织类型选项
export const ORG_TYPE_OPTIONS = [
  { label: '公司', value: 'company' },
  { label: '部门', value: 'department' },
  { label: '组', value: 'group' },
  { label: '岗位', value: 'position' }
]

/**
 * 获取组织树
 */
export function getOrganizationTree() {
  return api.get<Organization[]>('/organization/tree')
}

/**
 * 获取启用的组织树
 */
export function getActiveOrganizationTree() {
  return api.get<Organization[]>('/organization/tree/active')
}

/**
 * 获取组织列表
 */
export function getOrganizationList() {
  return api.get<Organization[]>('/organization/list')
}

/**
 * 获取组织详情
 */
export function getOrganizationById(id: number) {
  return api.get<Organization>(`/organization/${id}`)
}

/**
 * 获取子组织
 */
export function getOrganizationChildren(id: number) {
  return api.get<Organization[]>(`/organization/${id}/children`)
}

/**
 * 创建组织
 */
export function createOrganization(data: OrganizationForm) {
  return api.post<Organization>('/organization', data)
}

/**
 * 更新组织
 */
export function updateOrganization(id: number, data: OrganizationForm) {
  return api.put<Organization>(`/organization/${id}`, data)
}

/**
 * 删除组织
 */
export function deleteOrganization(id: number) {
  return api.delete<void>(`/organization/${id}`)
}

/**
 * 批量删除组织
 */
export function batchDeleteOrganizations(ids: number[]) {
  return api.delete<void>('/organization/batch', { data: ids })
}
